### ISSUE TITLE
pylint-protobuf raises astroid.exceptions.InferenceError if nothing is inferred

### ISSUE THREAD

**Issue by jckegelman** [nelfin/pylint-protobuf#5]
Hello, thanks for the plugin! It's working pretty well so far.

I came across an issue and was able to reproduce it by running `pylint` with the `pylint-protobuf` plugin on the following code:
```python
FILENAME = "/path:target.list"
BASENAME = FILENAME.split(":")[-1]
```

This raises the following exception(s):
```
Traceback (most recent call last):
  File ".../astroid/decorators.py", line 131, in raise_if_nothing_inferred
    yield next(generator)
StopIteration: {'node': <Call l.6 at 0x7f969e4ab748>, 'context': <astroid.context.InferenceContext object at 0x7f969dab45c8>}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".../pylint/__main__.py", line 7, in <module>
    pylint.run_pylint()
  File ".../pylint/__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File ".../pylint/lint.py", line 1628, in __init__
    linter.check(args)
  File ".../pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File ".../pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File ".../pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File ".../pylint/utils.py", line 1303, in walk
    self.walk(child)
  File ".../pylint/utils.py", line 1300, in walk
    cb(astroid)
  File ".../pylint_protobuf/__init__.py", line 437, in visit_assign
    self._visit_assign(node)
  File ".../pylint_protobuf/__init__.py", line 440, in _visit_assign
    new_scope, messages = visit_assign_node(self._scope, self._type_fields, node)
  File ".../pylint_protobuf/__init__.py", line 188, in visit_assign_node
    new_scope.update(_assign(old_scope, target, value))
  File ".../pylint_protobuf/__init__.py", line 145, in _assign
    new_type = _typeof(new_scope, rhs)
  File ".../pylint_protobuf/__init__.py", line 112, in _typeof
    return _typeof(scope, _slice(node))
  File ".../pylint_protobuf/__init__.py", line 44, in _slice
    indexable = next(value.infer())
  File ".../astroid/decorators.py", line 136, in raise_if_nothing_inferred
    raise exceptions.InferenceError(**error.args[0])
astroid.exceptions.InferenceError: Inference failed for <Call l.6 at 0x7f969e4ab748>.
```

It looks like the `_slice` method expects `infer` to sometimes return an `Uninferable` object, but instead astroid raises an `astroid.exceptions.InferenceError`.

https://github.com/nelfin/pylint-protobuf/blob/20b010032535047f24df9efbf88c28b0754ed0e1/pylint_protobuf/__init__.py#L44-L47

Maybe `_slice` could handle the exception and return `None` instead, something like:

```python
    try:
        indexable = next(value.infer())
        index = next(idx.infer())
    except astroid.exceptions.InferenceError:
        return None
```

What do you think? Would you like me to create a PR?

**Comment by nelfin** [comment]
Hi @jckegelman,

Thanks for the great issue and the test case. I'd be happy to merge a PR if
you have the time (otherwise I'll get to this asap).

**Comment by nelfin** [comment]
I've pushed a pre-release for this and a test case matching the snippet you added. If you can, try testing it out with `pip install pylint-protobuf==0.4.dev1` and see if it works with your environment. As an aside, it's not clear to me that `astroid` *should* be raising `InferenceError` at all, it seems to be an implementation detail that's leaking through.

**Comment by jckegelman** [comment]
that's great, thanks!
